### ISSUE TITLE
Still a problem on @microsoft/generator-sharepoint@1.13.1 > Gulp serve after file changes not getting reflected #5395

### ISSUE THREAD

**Issue by snugam** [SharePoint/sp-dev-docs#7638]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Developer environment

Windows

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [X] 💥 Microsoft Edge
- [ ] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

- browser version:  Microsoft Edge Version 97.0.1072.55 (Official build) (64-bit)
- SPFx version: @microsoft/generator-sharepoint@1.13.1
- Node.js version:  v14.15.0
- Windows 8.1

### Describe the bug / error

In a previous post (Gulp serve after file changes not getting reflected #5395) it was mentioned that this is now fixed -- however I am stilling getting the same problem unless I add "sessionStorage.clear();" 

### Steps to reproduce

1. Create an extension for SharePoint Online using yo
2. Write some code
3. execute gulp serve
4. Changes are visible the first time
5. make some more changes to the code and refresh the same page.
6. the changes are not visible unless ctrl+c is applied and gulp serve is restarted having to confirm/accept all the annoying debug/script warnings.

### Expected behavior

The browser page should refresh when pressing ctrl + r (with cache option disabled in devtools), instead of having to close the table make the changes and then running gulp serve again (this is not effective).

**Comment by AJIXuMuK** [comment]
Hi @snugam - just to confirm few things:
- you are creating a Command Set extension
- the extension is not installed to the app catalog, you're just serving it using debug query string from localhost
Is that correct?

Few more questions:
1. Are you seeing that the rebuild of the project is happening in the command prompt/terminal?
2. Do you have the cache enabled or disabled for the Network requests?

Thanks!

**Comment by snugam** [comment]
Hi there, 

Yes to your question "_the extension is not installed to the app catalog,_"
Yes to your question " _you're just serving it using debug query string from localhost Is that correct?_"

(to your few more Q.1) Yes, gulp serve is running, and you can see some "deployment" when I save the code.  It does also work fine when I add this line of code "sessionStorage.clear()".

(to your few more Q.2) Under devtools, cache is disabled (while devtools is open).

Hope it makes sense :)

**Comment by AJIXuMuK** [comment]
@snugam - could you check one more thing?
Will the issue disappear if you add `sw=bypass` parameter to the Query String?

**Comment by snugam** [comment]
I have removed `localStorage.clear();`.
Added `sw=bypass` to the query string.  
Reloading the page loads latest code changes.

When I counter test -
By removing `localStorage.clear();`.
Did not add `sw=bypass` to the query string.
Reloading the page does not load latest code changes.

I've also noticed that I have to re-add `sw=bypass` each time after reloading as it is removed after the page loaded.

Thanks again for your assistance!

**Comment by AJIXuMuK** [comment]
@snugam  - could you share your solution with me? As I can't repro the issue.
- I have created a Command Set solution using 1.13.1 generator
- `gulp serve`
- copied the debug string and pasted it to my list's page
- made changes back and forth (`alert` in `onInit`, `onListViewUpdated`)
- refreshed pages using f5, Ctrl+F5
The latest code is loaded correctly.
Am I missing something from your scenario? Maybe you have modifications in `config\serve.json`? For example, disabled `https` or default port instead of `4321`?

**Comment by snugam** [comment]
thank you -- I believe you, it could be that I am missing something.  Because I am running the dev work on a VM the `localStorage.clear()` does not present a big problem;  I have sent you a connection request on LinkedIn.

**Comment by snugam** [comment]
To be honest, I feel a little dumb here; perhaps others can also learn from this. 

I've removed the `localStorage.clear()` instruction saved the code, watched `gulp serve` do its _thing_; switched to the browser pressed CTRL+R to refresh AND... not new code. But, without making any changes, I thought, let me just check and see what happens if I refresh again and VOILA... the new change was there, without `localStorage.clear()`.  

I'm not sure how I managed to experience something very similar to this bug at the start only to discovery that I might have just been too quick in switching and refreshing back and forth.

thanks again @AJIXuMuK for the guidance!
